### ISSUE TITLE
UI errors since #1977

### ISSUE THREAD

**Issue by ysbaddaden** [instedd/surveda#2040]
In #1977 we started loading resources to render the current page _on-demand_, instead of loading all folders, surveys and panel surveys _upfront_ which was slow & bloated (even loading panel survey waves twice). This introduced a number of UI glitches or plain errors because the React UI continues to assume all resources are always loaded.

Here is a list of broken, or potentially broken operations:

- [x] at the root:
  - [x] delete a folder: it should disappear from view (see #2043);
  - [x] delete a survey: it should disappear from view (and pagination be OK);
  - [x] delete a panel survey: it should disappear from view (and pagination be OK);
  - [x] delete a panel survey wave: it should show the previous wave;
  - [x] move a survey to a folder: it should disappear from view (and pagination be OK);
  - [x] move a panel survey to a folder: it should disappear from view (and pagination be OK);
  - [x] rename a survey: returning to the root must show the new name;
  - [x] rename a panel survey wave: returning to the root must show the new name;
  - [x] rename a folder: it must show the new name (see #2043);
- [x] in a folder:
  - [x] delete a survey: it should disappear from view (and pagination be OK);
  - [x] delete a panel survey: it should disappear from view (and pagination be OK);
  - [x] delete a panel survey wave: it should show the previous wave;
  - [x] move a survey to another folder: it should disappear from view (and pagination be OK);
  - [x] move a survey to the root: it should disappear from view (and pagination be OK);
  - [x] move a panel survey to another folder: it should disappear from view (and pagination be OK);
  - [x] move a panel survey to the root: it should disappear from view (and pagination be OK);
  - [x] rename a survey: returning to the folder should show the new name;
- [x] in a panel survey:
  - [x] delete a wave: it should disappear from view (and pagination be OK);
  - [x] rename a wave: returning to the panel survey should show the new name;

AFAIK we can't rename a panel survey.

**Comment by ysbaddaden** [comment]
@mauri Here are some notes for the QA:

We refactored everything related to the navigation in folders, surveys and panel surveys, and the different operations on them (create, move, delete), change something in panel/surveys and see if they are reflected in the navigation (e.g. rename).

What has changed:

- We only load the folders, surveys & panel surveys necessary for the current view. We now properly take care to load what is needed on each view on-demand. I let some refresh issues slip, as well as this bug: renaming a folder when we didn't previously loaded the "root" of the project triggered an error (we expected folders to be loaded, but they were not).
- We refactored the UI to display folders, surveys, panel surveys, and waves inside a panel survey in a more consistent way. It broke a few operations (e.g. rename/delete a folder).

Sadly, we don't have automated tests to verify that nothing else broke.
